### PR TITLE
[BUG] twe alignment path fix and refactor

### PR DESCRIPTION
**Pull request by chrisholder**
#### Reference Issues/PRs

Fixed issue where twe path was one to long (as it is padded with 0s) so added seperate path for twe.

#### What does this implement/fix? Explain your changes.

Fixes twe alignment path and additionally simplfiies the code a bit with the refactor.

##### For all contributions
- [x] I've added myself to the [list of contributors](https://github.com/alan-turing-institute/sktime/blob/main/.all-contributorsrc).
- [x] Optionally, I've updated sktime's [CODEOWNERS](https://github.com/alan-turing-institute/sktime/blob/main/CODEOWNERS) to receive notifications about future changes to these files.
- [x] I've added unit tests and made sure they pass locally.
- [x] The PR title starts with either [ENH], [MNT], [DOC], or [BUG] indicating whether the PR topic is related to enhancement, maintenance, documentation, or bug.

##### For new estimators
- [x] I've added the estimator to the online documentation.
- [x] I've updated the existing example notebooks or provided a new one to showcase how my estimator works.

### CODE REVIEW

File: setup.cfg [hunk 1]
```diff
@@ -12,14 +12,14 @@ addopts =
...
-    --matrixdesign True
-    -n auto
+;    --durations 10
+;    --timeout 600
+;    --cov sktime
+;    --cov-report xml
+;    --cov-report html
+;    --showlocals
+;    --matrixdesign True
+;    -n auto
```
Comment (TonyBagnall) [inline setup.cfg:22]: is this PR meant to be changing this file?

File: sktime/distances/_distance_alignment_paths.py [hunk 2]
```diff
@@ -34,38 +34,99 @@ def compute_min_return_path(
...
-    while alignment[-1] != (0, 0):
-        i, j = alignment[-1]
-        if i == 0:
-            alignment.append((0, j - 1))
-        elif j == 0:
-            alignment.append((i - 1, 0))
+    i = x_size - 1
+    j = y_size - 1
+    alignment = []
+    while True:
```
Comment (TonyBagnall) [inline sktime/distances/_distance_alignment_paths.py:45]: stylistically, I'm not that keen on while True then breaking out of the loop

**Review by TonyBagnall — APPROVED** [review]

File: setup.cfg (hunk 1 above)
Comment (chrisholder) [reply setup.cfg:22]: Ah my mistake. I have to disable this to get pytest working on my local pc so Ill revert this

**Review by TonyBagnall — APPROVED** [review]

**Comment by fkiraly** [comment]
merging this - next time, please describe properly what issue is being fixed, @chrisholder. Optimally include a test.

**Pull request merged**